### PR TITLE
Feature/tooljetdb join operations

### LINKED ISSUE #7051: Add ability to join tables using tooljet database
### Is your proposal related to a problem?
There is no means to join tables created on tooljet database.

### Describe the solution you'd like
We should be able to join two tables based on two columns and a filter applied from the query manager.

### Describe alternatives you've considered
N/A

### If the feature is approved, would you be willing to submit a PR?
yes

### PR DESCRIPTION
**Pull request by ganesh8056**
PR For Tooljet DB Joins
User can Join tables, Filter and Sort the values in the Table

Closes #7051

### CODE REVIEW

File: frontend/src/Editor/QueryManager/QueryEditors/TooljetDatabase/ToolJetDbOperations.jsx [hunk 1]
```diff
@@ -33,24 +35,128 @@ const ToolJetDbOperations = ({ optionchanged, options, darkMode, isHorizontalLay
...
+          if (leftField?.table) {
+            currTableSet.add(leftField?.table);
+          }
+          if (rightField?.table) {
+            currTableSet.add(rightField?.table);
+          }
+        });
+      });
+      currTableSet.add(selectedTableId);
+      const newTables = difference([...tableSet], [...currTableSet]);
```
Comment (akshaysasidrn) [inline frontend/src/Editor/QueryManager/QueryEditors/TooljetDatabase/ToolJetDbOperations.jsx:89]: TIL, handy function!

File: frontend/src/Editor/QueryManager/QueryEditors/TooljetDatabase/util.js [hunk 2]
```diff
@@ -40,3 +40,27 @@ export const isOperatorOptions = [
   { value: 'null', label: 'null' },
   { value: 'notNull', label: 'not null' },
 ];
+
+// Discuss with Akshay whether in operator is needed, because we dont support array type
```
Comment (akshaysasidrn) [inline frontend/src/Editor/QueryManager/QueryEditors/TooljetDatabase/util.js:44]: can remove comment

**Review by akshaysasidrn — CHANGES_REQUESTED** [review]

File: frontend/src/Editor/QueryManager/QueryEditors/TooljetDatabase/DropDownSelect.jsx [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+          e.stopPropagation();
+        }}
+      >
+        <Remove fill="var(--slate12)" />
+      </span>
+    </Badge>
+  ));
+}
+
+function generateRandomId(length) {
```
Comment (akshaysasidrn) [inline frontend/src/Editor/QueryManager/QueryEditors/TooljetDatabase/DropDownSelect.jsx:207]: note, you can rather use `uuidv4` rather

File: server/src/controllers/tooljet_db.controller.ts [hunk 4]
```diff
@@ -97,4 +97,38 @@ export class TooljetDbController {
...
+          tableSet.add(leftField?.table);
+        }
+        if (rightField?.table) {
+          tableSet.add(rightField?.table);
+        }
+      });
+    });
+
+    const tables = [...tableSet].map((table) => ({
+      name: table,
```
[the remark this answers is not in the document]
Comment (ganesh8056) [reply server/src/controllers/tooljet_db.controller.ts:123]: But to change it everywhere would take lot of effort, need to discuss over huddle

File: frontend/src/Editor/QueryManager/QueryEditors/TooljetDatabase/DropDownSelect.jsx (hunk 3 above)
Comment (ganesh8056) [reply frontend/src/Editor/QueryManager/QueryEditors/TooljetDatabase/DropDownSelect.jsx:207]: Added in upcoming PR

**Pull request merged**